### PR TITLE
update to ensure that we have the correct payment amount at the tiem of execution, update the make payment computation to use the approval value instead of the CTBalance(ERC20 balance), this ensures correctness over what is allowed to be spent closes AST-1404, AST-1402

### PR DESCRIPTION
**Pull request by androolloyd**

### CODE REVIEW

**Comment by SantiagoGregory** [comment]
https://github.com/spearbit-audits/review-astaria2/issues/27
https://github.com/spearbit-audits/review-astaria2/issues/47

File: src/CollateralToken.sol [hunk 1]
```diff
@@ -157,6 +160,24 @@ contract CollateralToken is
...
+      );
+      if (
+        paymentToken.balanceOf(address(this)) < payment ||
+        payment !=
+        _locateCurrentAmount(
+          lia,
+          1000 wei,
+          zoneParameters.startTime,
+          zoneParameters.endTime,
+          true // round down
```
Comment (Saw-mon-and-Natalie) [inline src/CollateralToken.sol:175]: round up

Comment (androolloyd) [reply src/CollateralToken.sol:175]: Round down is false iirc.

Comment (Saw-mon-and-Natalie) [reply src/CollateralToken.sol:175]: what I mean was:

```suggestion
          true // round up
```

**Pull request merged**